src/Elastic.Apm/Config/AbstractConfigurationWithEnvFallbackReader.cs

```diff
@@ -41,6 +41,9 @@ internal AbstractConfigurationWithEnvFallbackReader(IApmLogger logger, string de
 		public virtual string Environment => ParseEnvironment(Read(ConfigConsts.KeyNames.Environment, ConfigConsts.EnvVarNames.Environment))
 			?? _defaultEnvironmentName;
 
+		public string ServiceNodeName => ParseServiceNodeName(Kv(ConfigConsts.EnvVarNames.ServiceNodeName,
```

Comment:
Here we force the `MicrosoftExtensionsConfig` to only read environment variables - what's the reason for this? I think it's reasonable that this setting can also be configured through a config file which contains different values for the different instances. Or did I miss something? 

I'd just do this here:

```
ParseServiceNodeName(Read(ConfigConsts.KeyNames.ServiceNodeName, ConfigConsts.EnvVarNames.ServiceNodeName));
```

_`ConfigConsts.KeyNames.ServiceNodeNames` also needs to be added in this case_

Reply:
done